shared/src/arguments.rs

```diff
@@ -51,6 +51,10 @@ pub struct Arguments {
     #[structopt(long, env = "UNSUPPORTED_TOKENS", use_delimiter = true)]
     pub unsupported_tokens: Vec<H160>,
 
+    /// Fee discount factor: 1 means no discount, 0.9 means 10% discount.
```

Comment:
I might elaborate here and say something about how this is a value in the range `[0, 1]`. Do we have any assertions that this value is `<= 1` anywhere?

Reply:
I actually don't really love this name, but since it was already around, I followed along. 

My understanding is that `fee_discount_factor` is a number, that multiplied by the fee we would charge for that order if not subsidizing anything, gives the effective fee that is charged to the user.

```
subsidized_fee = unsubsidized_fee * fee_discount_factor
```

That means the effective fee payed by the user goes down as fee_discount goes down, which is why I think calling it "discount" is misleading (we usually pay less when the discount is larger, and not the other way around).

If I got the above right, as far as I can tell, the domain for `subsidized_fee` is therefore `[0, +oo[` . As you point out, values greater than 1 don't make much sense, and it has the extra shortcoming that when we can't solve the equation for `unsubsidized_fee` when `fee_discount_factor=0`.

To solve all those problems, we could instead model it like this:

```
subsidized_fee = unsubsidized_fee * (1 - fee_discount_factor)
```

and require that `fee_discount_factor` is in `[0,1]` .

Reply:
Like you said Marco, it was already this way so no need to fix it in this PR. I do think what you say makes sense so I'm open to that change.